nixos/modules/services/networking/webhook.nix

```
@@ -0,0 +1,160 @@
+        '';
+      };
+      group = mkOption {
+        type = types.str;
+        default = defaultUser;
+        description = mdDoc ''
+          Webhook will be run under this group.
+
+          If set, you must create this group yourself!
+        '';
+      };
+      ip = mkOption {
+        type = types.str;
+        default = "0.0.0.0";
+        description = mdDoc ''
+          The IP webhook should serve hooks on.
+
+          The default means it be reached on any interface if `openFirewall = true`.
+          Otherwise, it can only be reached from the localhost.
+        '';
+      };
+      port = mkOption {
+        type = types.port;
+        default = 9000;
+        description = mdDoc "The port webhook should be reachable from.";
+      };
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Open the configured port in the firewall for external ingress traffic.
+          Preferably the Webhook server is instead put behind a reverse proxy.
+        '';
+      };
+      enableTemplate = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc "Enable the generated hooks file to be parsed as a Go template.";
+      };
+      urlPrefix = mkOption {
+        type = types.str;
+        default = "hooks";
+        description = mdDoc ''
+          The URL path prefix to use for served hooks
+
+          Example: protocol://yourserver:port/PREFIX/:hook-id
+        '';
+      };
+      hooks = mkOption {
+        type = types.str;
+        default = ''
+        [{
+          "id": "echo",
+          "execute-command": "echo",
+          "response-message": "Webhook is reachable!"
+        }]'';
+        description = mdDoc ''
+          The actual configuration of which hooks will be served.
+
+          It cannot be empty otherwise your service won't start.
+          Read more on the [project homepage] and on the [hook definition] page.
+
+          [hook definition]: https://github.com/adnanh/webhook/blob/master/docs/Hook-Definition.md
+          [project homepage]: https://github.com/adnanh/webhook#configuration
+        '';
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.users = mkIf (cfg.user == defaultUser) {
+      ${defaultUser} =
+        {
+          isSystemUser = true;
+          group = cfg.group;
+          description = "Webhook daemon user";
+        };
+    };
+
+    users.groups = mkIf (cfg.user == defaultUser && cfg.group == defaultUser) {
+      ${defaultUser} = {};
+    };
+
+    networking.firewall = lib.mkIf cfg.openFirewall {
+      allowedTCPPorts = [ cfg.port ];
+    };
+
+    systemd.services = {
+      webhook = {
+        description = "Webhook service";
+        after = [ "network.target" ];
+        environment = {
+          STNORESTART = "yes";
+          STNOUPGRADE = "yes";
+          inherit (cfg);
+        } // config.networking.proxy.envVars;
```

This looks like it was cargo-culted from the Syncthing service.

```suggestion
        environment = config.networking.proxy.envVars;
```

---

Probably was. Applying suggestion